es-ds-components/components/es-popover.vue

```diff
@@ -0,0 +1,157 @@
+                @click="closePanel">
+                <icon-x class="custom-x-ixon" height="20px" width="20px" />
+            </button>
+        </div>
+    </OverlayPanel>
+</template>
+<script setup lang="ts">
+import OverlayPanel from 'primevue/overlaypanel';
+import IconX from './icon/x.vue';
+
+const props = defineProps({
+    target: {
+        type: String,
+        required: true,
+    },
+    triggers: {
+        type: [String, Object],
+        required: false,
+        default: 'focus',
+    },
+    placement: {
+        type: String,
+        required: false,
+        default: 'auto',
+        validator: (val: string) => ['auto', 'top', 'bottom', 'right', 'left', 'topleft', 'topright', 'bottomleft', 'bottomright', 'lefttop', 'leftbottom', 'righttop', 'rightbottom'].includes(val),
+    },
+    show: {
+        type: Boolean,
+        required: false,
+        default: false,
+    },
+    variant: {
+        type: String,
+        required: false,
+        default: 'dark',
+        validator: (val: string) => ['light', 'dark'].includes(val as string),
+    },
+});
+
+const op = ref<InstanceType<typeof OverlayPanel> | null>(null);
+const slots = useSlots();
+const hasTitle = computed(() => !!slots.title);
+const closePanel = () => {
+    op.value?.hide();
+};
+const showPanel = (event: Event) => {
+    if (op.value) {
+        op.value.toggle(event);
+    }
+};
+
+
+onMounted(() => {
+    const targetElement = document.getElementById(props.target);
+    if (targetElement) {
+        targetElement.addEventListener(props.triggers as string, showPanel);
+    }
+});
+
+watch(
+    () => props.show,
+    (newVal) => {
+        if (newVal) {
+            const targetElement = document.getElementById(props.target);
+            if (targetElement) {
+                const event = new MouseEvent('click', {
+                    view: window,
+                    bubbles: true,
+                    cancelable: true,
+                });
+                targetElement.dispatchEvent(event);
+            }
+        } else {
+            closePanel();
+        }
+    }
+);
+
+const overlayPanelStyle = computed(() => {
+    switch (props.placement) {
+        case 'topright':
+            return {
+                transform: 'translateX(7%) translateY(-100%)',
+            };
+        case 'bottomright':
+            return {
+                transform: 'translateX(3%) translateY(0)',
+            };
+        case 'leftbottom':
+            return {
+                transform: 'translateX(-100%) translateY(0)',
+            };
+        case 'rightbottom':
+            return {
+                transform: 'translateX(7%) translateY(-50%)',
+            }
+        default:
+            return {};
+    }
+});
+
+const popoverPt = {
+    root: {
+         class: `popover ${props.variant === 'light' ? 'es-popover-light':'es-popover-dark'}`,
+    },
+    content: {
+        class: 'es-popover-close',
+    },
+};
+</script>
+
+
+<style lang="scss">
```

Comment:
It looks like the previous design included `hover: pointer`, we might want to add that here